Logic/logic_manager.cpp

```diff
@@ -8,46 +10,116 @@
-  ParserConnectionClient client(client_io_context, endpoints);
-  client_io_context.run();
+void raise_parser_connection_and_users_server() {
+  boost::asio::io_context global_io_context;
+  ip::tcp::resolver resolver(global_io_context);
+  auto endpoints = resolver.resolve(PARSER_IP, std::to_string(PARSER_PORT));
+  ParserConnectionClient client(global_io_context, endpoints);
+  auto users_server = UsersConnectionsServer(global_io_context);
+  users_server.listen(USERS_SERVER_PORT);
+  global_io_context.run();
 }
 
-// Запускаем в отдельном потоке из Main::run
-void raise_users_server(uint16_t users_server_port) {
-  // Поднимаем сервер для подключений клиентов:
-  auto users_server = UsersConnectionsServer();
-  users_server.listen(users_server_port);
-  users_server.run();
+void to_analysis() {
+  std::string response;
+  std::string last_response;
+  while (true) {
+    bool has_new_responses = false;
+    while (parsers_responses.try_dequeue(response)) {
+      last_response = response;
+      has_new_responses = true;
+    }
+    if (has_new_responses) {
+      if (PARSER_RESPONSE_LOGS_ON) {
+        std::cout << "Last response: " << last_response << std::endl;
+      }
+
+      Orders orders;
+      MarketRates market_rates;
+      if (!last_response.empty() && last_response[0] == '{') {
+        unpack_json(last_response, orders, market_rates);
+      } else {
+        if (PARSER_RESPONSE_LOGS_ON) {
+          std::cout << "Received empty response.\n";
+        }
+        continue;
+      }
+
+      Analysis analysis;
+      Chains chains;
+      analysis.analyze(chains, orders, market_rates);
+
+      if (CHAINS_LOGS_ON) {
+        std::cout << "Chains size: " << chains.list.size() << std::endl;
+        std::cout << "Chains:" << std::endl;
+        for (auto &chain : chains.list) {
+          std::cout << "Buy: " << chain.buy.coin2 << ", " << chain.change.first
+                    << " -> " << chain.change.second
+                    << ", Sell: " << chain.sell.coin2
+                    << ", Spread: " << chain.spread << "%" << std::endl;
+        }
+      }
+
+      up_to_date_version.set(pack_json(chains));
+
+    } else {
+      std::this_thread::sleep_for(
+          std::chrono::milliseconds(100)); // Пауза на 100 миллисекунд
+    }
+  }
 }
 
-struct Main {
-  // DataReceiver reciever;
-  // Analysis analysis;
-  // DataSender sender;
-
-  // Вероятно, оно будет не совсем так, но в упрощенном виде можно представлять
-  // что-то такое 
-  // void run(){
-  //     std::thread t1(&DataReceiver::receive, &reciever);
-  //     std::thread t2(&Analysis::analyze, &analysis);
-  //     std::thread t3(&DataSender::send, &sender);
-  //     std::thread t4(&DataSender::connection_handler, &sender)
-  // }
-};
+void apply_config() {
+  std::ifstream config("config.json");
+  nlohmann::json j;
+  config >> j;
+  config.close();
+  PARSER_IP = j["endpoints"]["parser's ip"].get<std::string>();
```

Comment:
Instead of manual parsing of options you could've used a library like `CLI11` that would allow you to use them as CLI params as well as json config file.